pkg/typesystem/typesystem.go

```diff
@@ -1203,25 +1202,33 @@ func (t *TypeSystem) IsTuplesetRelation(objectType, relation string) (bool, erro
 	return false, nil
 }
 
-func tupleToUsersetsDefinitions(relationDef *openfgav1.Userset, resp *[]*openfgav1.TupleToUserset) []*openfgav1.TupleToUserset {
-	if relationDef.GetTupleToUserset() != nil {
-		*resp = append(*resp, relationDef.GetTupleToUserset())
-	}
-	if relationDef.GetUnion() != nil {
-		for _, child := range relationDef.GetUnion().GetChild() {
-			tupleToUsersetsDefinitions(child, resp)
+func flattenUserset(relationDef *openfgav1.Userset) []*openfgav1.TupleToUserset {
+	output := make([]*openfgav1.TupleToUserset, 0)
+	userset := relationDef.GetUserset()
+	switch x := userset.(type) {
+	case *openfgav1.Userset_TupleToUserset:
+		if x.TupleToUserset != nil {
+			output = append(output, x.TupleToUserset)
 		}
-	}
-	if relationDef.GetIntersection() != nil {
-		for _, child := range relationDef.GetIntersection().GetChild() {
-			tupleToUsersetsDefinitions(child, resp)
+	case *openfgav1.Userset_Union:
```

Comment:
should we not have a case for `computedUserset` too? 😬

Reply:
I was thinking the same thing, but didn't know for sure

Reply:
You are right, there are certainly better ways of handling this. I have a hunch that I may be able to reuse some existing code to get this answer more efficiently. Stay tuned.

Reply:
Don't have to do this now, open an issue 👍

Reply:
I don't see any reason why we can't support `parent` being a computed userset as long as it resolves to a direct relation. Not saying that we should resolve it in this PR but perhaps a nice to have. cc @rhamzeh